### PR TITLE
Qgb universal nonces ADR update

### PR DESCRIPTION
**Pull request by SweeXordious**
Closing https://github.com/celestiaorg/celestia-app/issues/471

### CODE REVIEW

File: docs/architecture/ADR-004-QGB_relayer_security.md [hunk 1]
```diff
@@ -85,11 +85,317 @@ A proposition to remediate the issues described above is to make the signatures
...
+    dataCommitment := types.NewDataCommitment(nonce, beginBlock, endBlock)
+    return *dataCommitment, nil
+}
+```
+
+And store it.
+
+#### handleValsetRequest
+
+The `handleValsetRequest` is more involving as it has more criteria to create new valsets:
```
Comment (rootulp) [inline docs/architecture/ADR-004-QGB_relayer_security.md:334]:
```suggestion
The `handleValsetRequest` is more involved as it has more criteria to create new valsets:
```

**Review by rootulp — APPROVED** [review]
Thanks for publishing this ADR! It really helps clarify the implementation details behind universal nonces.

All my feedback is optional and non-blocking

File: docs/architecture/ADR-004-QGB_relayer_security.md [hunk 2]
```diff
@@ -85,11 +85,317 @@ A proposition to remediate the issues described above is to make the signatures
...
+On the state machine, we will need to store the attestations when needed. To do so, we will define the following:
+
+##### Store latest nonce
+
+We will need to keep track of the latest nonce to enforce the nonces order and avoid overwriting existing attestations. This will be done using the following:
+
+```go
+func (k Keeper) SetLatestAttestationNonce(ctx sdk.Context, nonce uint64) {
+    if k.CheckLatestAttestationNonce(ctx) && k.GetLatestAttestationNonce(ctx)+1 != nonce {
+        panic("not incrementing latest attestation nonce correctly!")
```
[the remark this answers is not in the document]
Comment (SweeXordious) [reply docs/architecture/ADR-004-QGB_relayer_security.md:194]: https://github.com/celestiaorg/celestia-app/issues/752

File: docs/architecture/ADR-004-QGB_relayer_security.md [hunk 3]
```diff
@@ -85,11 +85,317 @@ A proposition to remediate the issues described above is to make the signatures
...
+
+```go
+func (k Keeper) CheckLatestAttestationNonce(ctx sdk.Context) bool {
+    store := ctx.KVStore(k.storeKey)
+    has := store.Has([]byte(types.LatestAttestationtNonce))
+    return has
+}
+```
+
+- The provided nonce is different than `Latest nonce + 1`.
```
[the remark this answers is not in the document]
Comment (SweeXordious) [reply docs/architecture/ADR-004-QGB_relayer_security.md:214]: https://github.com/celestiaorg/celestia-app/issues/752

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**